ethcore/src/account.rs

```diff
@@ -143,13 +155,13 @@ impl Account {
 
 	/// Set (and cache) the contents of the trie's storage at `key` to `value`.
 	pub fn set_storage(&mut self, key: H256, value: H256) {
-		match self.storage_overlay.borrow_mut().entry(key) {
-			Entry::Occupied(ref mut entry) if entry.get().1 != value => {
-				entry.insert((Filth::Dirty, value));
+		match self.storage_changes.entry(key) {
```

Comment:
It may happen and is supported. `storage_cache` reflects the contents of the disk database,  `storage_changes` keeps changes over this contents. I'd prefer caching and tracking changes to be split into different structs, so that only the state can have changes. This is left for a future PR.